ros_buildfarm/common.py

```diff
@@ -533,3 +534,12 @@ def get_system_architecture():
     if machine == 'aarch64':
         return 'armv8'
     raise RuntimeError('Unable to determine architecture')
+
+
+def has_gpu_support():
+    # It detects only nvidia support. The implementation is to check if
+    # /dev/nvidia* links are present. Further work could be to use a
+    # more robust method (i.e: https://github.com/ntpeters/pyvidia)
+    if glob.glob('/dev/nvidia*'):
```

Comment:
If this check runs inside a non-nvidia runtime docker container (as I expect it would during the create_devel_task then the /dev/nvidia* devices aren't available).

Reply:
That is true. I think that this situation may refer to the use of [devel_create_tasks.Dockerfile.em](https://github.com/ros-infrastructure/ros_buildfarm/pull/624/files#diff-d2dacd6d9707bab9ae565f1625d5ecf2R21). If I understand correctly the nvidia-docker support can be removed from there since it's not being used (not necessary). Is there any more places in these changes where this function is called as part of the `create_devel_task` process?

Reply:
>  If I understand correctly the nvidia-docker support can be removed from there since it's not being used (not necessary)

I think that optimally, the devel_create_tasks container _shouldn't_ need the nvidia runtime. But to accomplish that I think we need to run `has_gpu_support()` on from the host system and pass  an argument through a command-line argument through to the build and test containers.

Reply:
That is true. I think that this situation may refer to the use of [devel_create_tasks.Dockerfile.em](https://github.com/ros-infrastructure/ros_buildfarm/pull/624/files#diff-d2dacd6d9707bab9ae565f1625d5ecf2R21). If I understand correctly the nvidia-docker support can be removed from there since it's not being used (not necessary). Is there any more places in these changes where this function is called as part of the `create_devel_task` process?

Reply:
Following this up and reviewing my old changes:
 1. The [current implementation](https://github.com/ros-infrastructure/ros_buildfarm/pull/624/files#diff-5cf979d1d9687b7cad0d95e630cc87c6R190) delegates the check to run docker with appropriate arguments to the runtime execution of the `devel_job.xml` . It's not very elegant but it is probably the best moment to check for the gpu support.
 1. The PR still hosts the support to detect the gpu in the devel_create_tasks container. If 1. is good, I think we can remove it. If 1. is no good then I can add support to pass the value of `has_gpu_support` to scripts in the form of a parameter.

@nuclearsandwich what do you think?